### DISCUSSION TITLE
Game engine prohibits _ character in headers

### DISCUSSION THREAD

**Discussion by smellynailclippings** [chatwoot/chatwoot#4793] [Chatwoot APIs]
Hello there,

I was curious as to whether or not there were any plans in the future to introduce an alternate authentication header. Based on what I have seen in other forums so far, the "api_access_token" header has been a problem for a couple other people.

For context, the game engine I am using blocks any outgoing HTTP requests with certain characters in them, _ being one of them. This makes authenticating requests to ChatWoot's API impossible.

![image](https://user-images.githubusercontent.com/63950233/171990581-ab038626-e112-407c-9c23-99ffcabda0d4.png)

**Comment by vishnu-narayanan** [comment]
@smellynailclippings Underscores(`_`) in HTTP headers are perfectly valid according to HTTP standards. Could you please check if it is possible to allow `_` with the game engine you are using?

For example, `nginx` has `underscores_in_headers on;` directive to enable underscores in headers.

**Reply by sojan-official** [reply]
we are also tracking https://github.com/chatwoot/chatwoot/issues/1615

**Reply by smellynailclippings** [reply]
For whatever reason they have given absolutely no option to allow the _ character in the headers of outgoing HTTP requests despite many requests for that feature.

**Reply by sojan-official** [reply]
@smellynailclippings  we will look at having an option to support an alternative HTTP header without `_` as a fallback. please follow the updates over #1615
